sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -600,3 +604,146 @@ case class Logarithm(left: Expression, right: Expression)
     """
   }
 }
+
+case class Round(child: Expression, scale: Expression) extends Expression with ExpectsInputTypes {
+
+  def this(child: Expression) = {
+    this(child, Literal(0))
+  }
+
+  override def children: Seq[Expression] = Seq(child, scale)
+
+  override def nullable: Boolean = true
+
+  override def foldable: Boolean = child.foldable
+
+  override lazy val dataType: DataType = child.dataType match {
+      case DecimalType.Fixed(p, s) => DecimalType(p, _scale)
+      case t => t
+    }
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(
+    // rely on precedence to implicit cast String into Double
+    TypeCollection(DecimalType, DoubleType, FloatType, LongType, IntegerType, ShortType, ByteType),
+    TypeCollection(LongType, IntegerType, ShortType, ByteType))
```

Comment:
the 2nd argument should just be an IntegerType. It will get implicitly casted anyway.